### ISSUE TITLE
When i download game files i get Error : Invalid Archive

### ISSUE THREAD

**Issue by exaker** [OpenRA/OpenRA#12985]
When i download game files i get Error : Invalid Archive so whats the problem guys? i`m getting this for like 1 year whenever i come back to game...

**Comment by reaperrr** [comment]
Where are you downloading it from?

**Comment by exaker** [comment]
Official Website

**Comment by reaperrr** [comment]
What OS do you have and which package did you download?
There haven't been many other reports of such errors (I don't know of any), so we need more details to figure out what the problem might be.

**Comment by exaker** [comment]
I have Win7, I downloaded Quick Download Red Alert 1 Which is 14mb or smth like that

**Comment by reaperrr** [comment]
I can't find anything called "Quick Download Red Alert 1" on our download page.

Are you sure you downloaded it from here? http://www.openra.net/download/

**Comment by pchote** [comment]
I assume this is about the ingame asset downloads.

**Comment by reaperrr** [comment]
Oh, I see.

**Comment by Snurre86** [comment]
Could you take a screenshot? What is "getting this for like 1 year" If you have over a year old version you can download the new release [here](http://www.openra.net/download/).

**Comment by exaker** [comment]
No i have downloaded newest playtest.

**Comment by exaker** [comment]
Ops i mean newest Release sorry

**Comment by Snurre86** [comment]
Make a new folder in the install folder "OpenRA" and call it Support "OpenRA\Support" then see what happens.

**Comment by exaker** [comment]
Nothing changed.

**Comment by abcdefg30** [comment]
Do you got any error logs to share (especially `install.log`)? They should be located at
`%USERPROFILE%\Documents\OpenRA\Logs` or inside `OpenRA\Support\Logs`.

**Comment by Snurre86** [comment]
Open `install.log` with a text editor and add the text on here as the file type can't be uploaded here or paste the text to [http://pastebin.com/](http://pastebin.com/) then place the link here.

**Comment by raigoinabox** [comment]
Hello, I am having the same problem. My OS is Ubuntu 16.04 and the openra package is the latest release package (20161019.release). This is the contents of my `install.log` after a failed download:
```
Fetching mirrors from http://www.openra.net/packages/ra-quickinstall-mirrors.txt
Downloading http://openra.mirror.haffdata.com/ra-quickinstall.zip
Extraction failed
Deleting temporary file /tmp/5s3oy55r.mfe
Download failed: Invalid archive
```

**Comment by reaperrr** [comment]
Hm, I just tried to reproduce this, but both on bleed and release 20161019 quickinstall worked fine.
At least the file on the haffdata mirror is not corrupted according to my tests, so the problem must occur either during download or extraction.

**Comment by raigoinabox** [comment]
So, I downloaded the source code and I noticed that "Invalid archive" is shown on **any** exception. I added a simple logging of that unknown exception, built and ran and I got this:
```
Fetching mirrors from http://www.openra.net/packages/ra-quickinstall-mirrors.txt
Downloading http://galaxy.republic.community/h/openra/ra-quickinstall.zip
System.NotSupportedException: No data is available for encoding 775.
  at System.Text.Encoding.GetEncoding (Int32 codepage) <0x7fb25106bd80 + 0x00360> in <filename unknown>:0 
  at ICSharpCode.SharpZipLib.Zip.ZipConstants.ConvertToString (System.Byte[] data, Int32 count) <0x40f374c0 + 0x00038> in <filename unknown>:0 
  at ICSharpCode.SharpZipLib.Zip.ZipConstants.ConvertToStringExt (Int32 flags, System.Byte[] data, Int32 count) <0x40f373e0 + 0x00063> in <filename unknown>:0 
  at ICSharpCode.SharpZipLib.Zip.ZipFile.ReadEntries () <0x40f35fe0 + 0x00673> in <filename unknown>:0 
  at ICSharpCode.SharpZipLib.Zip.ZipFile..ctor (System.IO.FileStream file) <0x40f359e0 + 0x0015f> in <filename unknown>:0 
Extraction failed
Deleting temporary file /tmp/vhyohnya.qwi
Download failed: Invalid archive
```
Very short googling got me that maybe this has something to with me having a Baltic locale or encoding? https://forums.xamarin.com/discussion/61880/system-notsupportedexception-regarding-encoding. This is the result of running on my machine `locale`:
```
LANG=et_EE.UTF-8
LANGUAGE=et
LC_CTYPE="et_EE.UTF-8"
LC_NUMERIC="et_EE.UTF-8"
LC_TIME="et_EE.UTF-8"
LC_COLLATE="et_EE.UTF-8"
LC_MONETARY="et_EE.UTF-8"
LC_MESSAGES="et_EE.UTF-8"
LC_PAPER="et_EE.UTF-8"
LC_NAME="et_EE.UTF-8"
LC_ADDRESS="et_EE.UTF-8"
LC_TELEPHONE="et_EE.UTF-8"
LC_MEASUREMENT="et_EE.UTF-8"
LC_IDENTIFICATION="et_EE.UTF-8"
LC_ALL=
```
In short this means that my locale is set to Estonia and my encoding is UTF8.

**Comment by vaincz** [comment]
raigoinabox: thank you for tip. I had the same problem and when I read your comment I just tried
```
export LC_ALL=en_US.UTF-8
export LANG=en_US.UTF-8
export LANGUAGE=en_US.UTF-8
```
and suddenly it works!

**Comment by rob-v** [comment]
Thanks @raigoinabox. This small PR #13119  doesn't solve the main issue but your reported issue: Add exception info to logged 'Extraction failed' (Invalid archive) error

**Comment by rob-v** [comment]
Workaround - Change temporarily your System locale: https://www.tenforums.com/general-support/20884-how-do-you-change-language-settings-non-unicode-programs-win10.html
